File: 167. Two Sum II - Input Array Is Sorted/twoSum.ts
function twoSum(numbers: number[], target: number): number[] {
    const storage = new Map()
    for (let i = 0; i < numbers.length; i++) {
        let x = target - numbers[i];
        if (storage.has(x)) {
            return[storage.get(x), i + 1]
        }
        storage.set(numbers[i], i + 1)
    }
    return []
};

function twoSum2(numbers: number[], target: number): number[] {
    let left = 0;
    let right = numbers.length - 1;
    while (left < right) {
        let sum = numbers[left] + numbers[right]
        if (target === sum) return [left + 1, right + 1]
        if (target > sum) left++
        else right--
    }
    return []
};

File: 167. Two Sum II - Input Array Is Sorted/twoSum.py
def two_sum(numbers: list[int], target: int) -> list[int]:
    seen = {}
    for i, num in enumerate(numbers):
        need = target - num
        if need in seen:
            return [seen[need], i + 1]
        seen[num] = i + 1
    return []


def two_sum_2(numbers: list[int], target: int) -> list[int]:
    left, right = 0, len(numbers) - 1
    while left < right:
        total = numbers[left] + numbers[right]
        if total == target:
            return [left + 1, right + 1]
        if total > target:
            right -= 1
        else:
            left += 1
    return []